content/news/XXXX-XX-XX-build-infrastructure-updates.md

```diff
@@ -0,0 +1,75 @@
+
+A great aspect of using GitHub Actions for our builds is that GitHub does the work of keeping the operating systems and build tools in the virtual machine images up to date for us. However, when we tried to build Mixxx with our old archive of dependencies on GitHub Actions, it would not build because GitHub Actions no longer had a version of XCode old enough to support the macOS 10.13 SDK which was used on our old macOS build server.
+
+So, we got GitHub Actions to run the scripts we used on the old build server to build all of Mixxx's dependencies. This took another about another week of work. With a new archive of dependencies built by GitHub Actions, we could build Mixxx on GitHub Actions. Then, more complications became apparent.
+
+The builds from GitHub Actions were so laggy that they were completely unusable, but the builds from the old build server with the same code were okay. It turned out that having our own build server that was such a hassle to upgrade that we didn't do it for years was masking a serious bug. Building Mixxx with a macOS SDK 10.14 or newer made Qt use layer backing. Combined with the legacy QGLWidget API we still use for Mixxx's waveforms, this was extremely slow. We have [previously attempted](https://github.com/mixxxdj/mixxx/pull/1974) to update to the newer QOpenGLWidget API. Although this performed much better on macOS, the waveforms jerked back and forth instead of scrolling smoothly. We considered trying to finish this work for 2.3, but the task is difficult and would require major changes to the waveform rendering code. Ultimately solving this performance issue will require [rewriting our whole GUI for Qt6](https://forum.qt.io/topic/121746/determing-graphics-frame-swap-time-when-rendering-in-qt-6/) which will also be a huge challenge. If anyone reading this is skilled at graphics programming, please get in touch on our [Zulip chat](https://mixxx.zulipchat.com/). We could really use help figuring out how to rewrite waveform rendering for Qt6. For now, we are using an ugly hack of automatically downloading the macOS 10.13 SDK to build Mixxx with current versions of XCode that do not officially support that old of a macOS SDK.
```

Comment:
Can you add a link that explains what layer backing is?

Btw, it's much easier to comment on individual sentences than whole paragraphs. For future PRs I recommend putting each sentence into a separate line (this also makes the diffs nicer). The final website post will look the same.

Reply:
If I had a link that explained wtf layer backing is, maybe we wouldn't be in this mess.

Reply:
> Btw, it's much easier to comment on individual sentences than whole paragraphs. For future PRs I recommend putting each sentence into a separate line (this also makes the diffs nicer). The final website post will look the same.

It is more cumbersome to write prose that way though.

Reply:
I usually write and then do `s/. /.\n/` afterwards :D